model/src/payout_curve.rs

```diff
@@ -105,27 +104,35 @@ fn calculate_payout_parameters(
             let long_amount_btc = row[2];
 
             let long_amount = to_sats(long_amount_btc)?;
+            let short_amount = to_sats(payout_curve.total_value - long_amount_btc)?;
 
-            let long_amount_adjusted = match fee {
-                // We use `saturating_sub` because the adjusted payout
-                // cannot go below zero. If the original payout is
-                // close or equal to zero and the fee is sufficiently
-                // large we would overflow otherwise.
-                FeeFlow::LongPaysShort(fee) => long_amount.saturating_sub(fee.as_sat()),
-                FeeFlow::ShortPaysLong(fee) => long_amount.add(fee.as_sat()),
-                FeeFlow::Nein => long_amount,
+            // We use `saturating_sub` when deducting fees because the
+            // adjusted payout cannot go below zero. If the original
+            // payout is close or equal to zero and the fee is
+            // sufficiently large we would overflow otherwise.
+            let (short_amount, long_amount) = match fee {
+                FeeFlow::LongPaysShort(fee) => {
+                    let long_minus_fee = long_amount.saturating_sub(fee.as_sat());
+                    let long_fee_deduction = (long_amount as i64) - (long_minus_fee as i64);
+                    let short_plus_fee = (short_amount as i64) + long_fee_deduction;
```

Comment:
We do a saturating sub so we have a bound for the lower end (cannot go below zero), but we don't have an upper bound when adding fees. 

Don't we have to cap the maximum payout to `long_margin` + `short_margin` here? 
I feel with this change it would be possible to have a higher payout then the actual amount available.

Reply:
Math checks out for me:

```markdown
pool = long + short

long_minus_fee = max(long - fee, 0)                     // capped because we can't go below 0

effective_fee = long - long_minus_fee
long_minus_fee = long - effective_fee                   // after rearranging

short_plus_fee = short + effective_fee

pool = long_minus_fee + short_plus_fee 
     = long - effective_fee + short + effective_fee     // substituting
     = long + short                                     // cancelling
```

At the end we get the same equation we started with.

Reply:
I think it does - I did not think it through correctly. 
I'll do it one more time in words :)

In a `LongPaysShort` scenario the fee goes from long to short, i.e. the fee will always be deducted from the long margin. This is what `long_minus_fee` represents, which is bounded to `[0,long_amount-1sat]`.

That means `long_fee_deduction` is bounded to `[long_amount,1sat]`.
That means `short_plus_fee` is bounded to `[short_amount + long_amount,short_amount+1sat]`.

Checks out :)